From 58d0ba48c37eb44da1dafb5c6c9241c0c8978666 Mon Sep 17 00:00:00 2001
From: squimrel <squimrel@users.noreply.github.com>
Date: Sun, 18 Jun 2017 23:05:19 +0000
Subject: [PATCH 09/40] Always add partition to drive

---
 helper/write.cpp | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/helper/write.cpp b/helper/write.cpp
index 0cf563b..3116f38 100644
--- a/helper/write.cpp
+++ b/helper/write.cpp
@@ -343,20 +343,21 @@ void write(const QString &source, Drive *const drive, bool persistentStorage) {
         writeCompressed(source, drive);
     else
         writePlain(source, drive);
+    drive->open();
     check(drive->getDescriptor());
+    drive->close();
+    drive->umount();
+    auto size = QFileInfo(source).size();
+    auto partitionInfo = drive->addPartition(size, partitionLabel);
     if (persistentStorage) {
-        drive->umount();
-        auto size = QFileInfo(source).size();
-        auto partitionInfo = drive->addPartition(size, partitionLabel);
         QString mountpoint = drive->mount(partitionInfo.first);
         zeroFile(mountpoint + overlayFilename, partitionInfo.second);
         drive->umount();
-        drive->open();
-        out.flush();
-        char *errstr;
-        if (implantISOFD(drive->getDescriptor(), false, true, true, &errstr) != 0) {
-            throw std::runtime_error(errstr);
-        }
-        drive->close();
     }
+    drive->open();
+    char *errstr;
+    if (implantISOFD(drive->getDescriptor(), false, true, true, &errstr) != 0) {
+        throw std::runtime_error(std::string(errstr));
+    }
+    drive->close();
 }
-- 
2.14.1

